pulsar-client-cpp/lib/OpSendMsg.h

```diff
@@ -25,9 +25,26 @@
 
 #include "TimeUtils.h"
 #include "MessageImpl.h"
+#include "ChunkMessageIdImpl.h"
 
 namespace pulsar {
 
+struct ProducerChunkedMessageCtx {
+   public:
+    ProducerChunkedMessageCtx() = default;
+
+    ProducerChunkedMessageCtx(MessageIdImplPtr firstChunkMessageId, MessageIdImplPtr lastChunkMessageId)
```

Comment:
Even it's a shared pointer, please pass it by const reference. (not only here)

See https://stackoverflow.com/questions/3310737/should-we-pass-a-shared-ptr-by-reference-or-by-value